libraries/c_sdk/standard/mqtt/src/iot_mqtt_operation.c

```diff
@@ -1150,7 +1150,7 @@ void _IotMqtt_ProcessOperation( _mqttOperation_t * pOperation )
     {
         /* Decrement reference count to signal completion of send job. Check
          * if the operation should be destroyed. */
-        if( waitable == true )
+        if( ( waitable == true ) && ( pOperation->u.operation.retry.limit <= 0 ) )
```

Comment:
why is this new condition required here?

Reply:
This condition is required to handle the publish with retries.

Reply:
As discussed, this condition may not be needed any more. Please update.